examples/preprocessing/plot_scaling_importance.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+=========================================================
+Importance of Feature Scaling
+=========================================================
+
+Features scaling though standardization (or Z-score normalization)
+can be an important preprocessing step for many machine learning
+algorithms. Standardization involves rescaling the features such
+that they’ll have the properties of a standard normal distribution
+with a mean of zero and a standard deviation of one.
+
+While many algorithms (such as SVM, K-nearest neighbors and logistic
+regression) require features to be normalized, intuitively we can
+think of Principle Component Analysis (PCA) as being a prime example
+of when normalization is important. In PCA we are interested in the
+components that maximize the variance. If there exists components
+(e.g human height) that vary less than other components (e.g human
+weight) because of their respective scales (meters vs. kilos) it can
+be seen how not scaling the features would cause PCA to determine that
+the direction of maximal variance more closely corresponds with the
+'weight' axis. As a change in height of one meter can be considered much
+more important than the change in weight of one kilogram, it is easily
+seen that this determination is incorrect. In the case of PCA, scaling
+features using normalization is preferred over using min-max scaling as
+the primary components are computed using the correlation matrix as
+opposed to the covariance matrix.
+
+In order to illustrate this in an example, PCA will be performed on a
+dataset which has been standardized using Standard Scaler and a copy
+which has remained untouched. The results with be visualized and a clear
+difference noted. The 1st principal component in the unscaled set is shown.
+It can be seen that feature #13 dominates the direction, being a whole
+two orders of magnitude above the other features. This is contrasted when
+observing the prinicpal component for the scaled version of the data. In the
+scaled version, the orders of magnitude is roughly the same across all the
+features.
+
+The results will then be used to train a naive Bayes classifier, and a clear
+difference in prediction accuracies will be observed.
```

Comment:
"wherein the dataset which is scaled before PCA vastly outperforms the unscaled version."